include/notcurses/notcurses.h

```diff
@@ -2423,7 +2423,7 @@ ncplane_pututf32_yx(struct ncplane* n, int y, int x, uint32_t u){
 
 static inline int
 ncplane_putwc_yx(struct ncplane* n, int y, int x, wchar_t w){
-  return ncplane_pututf32_yx(n, y, x, w);
+  return ncplane_pututf32_yx(n, y, x, (uint32_t)w);
```

Comment:
I'm not very familiar with wide characters.  Is it safe to assume the MSB is zero?

Reply:
nope. assume a wide character can be any 32-bit value unless you have very good reason to think otherwise.

Reply:
OK, in general that makes sense.  The arg to `ncplane_pututf32_yx` is `uint32_t` though so it's going to get converted anyway, with or without the warning.  And a negative `wchar_t` value won't map to a valid code point in UTF-32, whether you interpret it as signed or unsigned, right?

Reply:
it would depend on the value. `wchar_t` is weird. it does *not* necessarily correspond to UTF-32 (indeed, on Windows and Java it's only 16 bits). you might need more than one `wchar_t` to represent a given unicode character, let alone an EGC. you generally want to work with `wchar_t` as little as possible, converting everything to UTF-8 (which is everywhere `char *`). i only keep these APIs around to support the old-skool 1994 C APIs, which require some `wchar_t`.

the least complex way i can precisely answer your question is: if you pass a `wchar_t` which cannot be converted to a UTF-8 multibyte string on the local platform (as `wchar_t` is inherently a platform-specific deal), i will return failure.

Reply:
Yeah, I've run into 16-bit ur-Unicode in JVMland.

Anyway my change here doesn't change the meaning of the code; it just silences a warning.  If this cast isn't correct, there is a preexisting problem with this call, I believe.

Reply:
oh sure i was just holding off until you'd had all your questions answered